syntax = "proto3";
enum MovieCategory {
    MYSTERY = 0;
    SCIENCE_FICTION = 1;
    COMEDY = 2;
    THRILLER = 3;
    ACTION = 4;
	ALL = 5;
}

message RecommendationRequest {
    int32 user_id = 1;
    MovieCategory category = 2;
    int32 max_results = 5;
}

message MovieRecommendation {
    int32 id = 1;
    string title = 2;
}
message RecommendationResponse {
    repeated MovieRecommendation recommendations = 1;
}

service Recommendations {
    rpc Recommend (RecommendationRequest) returns (RecommendationResponse);
}
